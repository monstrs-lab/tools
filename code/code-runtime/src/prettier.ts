import { dymanicRequire } from './utils'

export const prettier = dymanicRequire.resolve('prettier')
export const prettierConfig = dymanicRequire.resolve('@monstrs/prettier-config')
