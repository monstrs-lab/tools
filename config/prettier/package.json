{
  "name": "@monstrs/config-prettier",
  "version": "0.0.14",
  "license": "MIT",
  "main": "src/index.ts",
  "files": [
    "dist"
  ],
  "scripts": {
    "build": "yarn library build",
    "prepack": "yarn run build",
    "postpack": "rm -rf dist"
  },
  "dependencies": {
    "prettier": "^2.8.4"
  },
  "devDependencies": {
    "@types/node": "^18.14.4"
  },
  "publishConfig": {
    "main": "dist/index.js",
    "typings": "dist/index.d.ts"
  }
}
