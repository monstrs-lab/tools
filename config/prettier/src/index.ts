export default {
  semi: false,
  singleQuote: true,
  tabWidth: 2,
  jsxSingleQuote: true,
  trailingComma: 'es5',
  plugins: [
    require.resolve('@monstrs/prettier-typescript-plugin'),
    require.resolve('prettier-plugin-packagejson'),
  ],
  printWidth: 100,
}
