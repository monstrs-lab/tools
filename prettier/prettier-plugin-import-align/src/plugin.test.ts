import { format } from 'prettier'

describe('import-align', () => {
  it('should match latest snapshot', () => {
    const source = `
import long from './long'
import a from './a'
      `

    const code = format(source, {
      parser: 'typescript',
      plugins: [require.resolve('../')],
    } as any)

    expect(code).toMatchSnapshot()
  })
})
