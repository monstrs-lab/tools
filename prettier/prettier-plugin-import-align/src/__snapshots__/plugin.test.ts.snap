// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`import-align should match latest snapshot 1`] = `
"import long from \\"./long\\";
import a    from \\"./a\\";
"
`;
